api/src/modules/timers/services/timer-engine.service.ts

```
@@ -48,24 +37,434 @@ export class TimerEngineService {
+        const timerConfig = stepInstance.timerConfig;
+        if (timerConfig.type === 'NO_TIMER') {
+            throw new BadRequestException('Cannot start timer for NO_TIMER type');
+        }
+
+        let allocatedTimeMs: number | null = null;
+        let scheduledEndAt: Date | null = null;
+
+        // 4. Calculate timer parameters based on type
+        switch (timerConfig.type) {
+            case 'FIXED_DURATION':
+                const durationHours = customDurationHours || timerConfig.durationHours;
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)